interpreter/exec/eval.ml

```diff
@@ -319,18 +323,28 @@ let rec step_thread (t : thread) : thread =
         (try
           assert (sz = None);
           check_align addr ty sz e.at;
+          check_shared mem e.at;
           let v = Memory.load_value mem addr offset ty in
           if v = ve then
             assert false  (* TODO *)
           else
             I32 1l :: vs', []  (* Not equal *)
         with exn -> vs', [Trapping (memory_error e.at exn) @@ e.at])
 
-      | MemoryAtomicNotify x, I32 count :: I32 i :: vs' ->
+      | MemoryAtomicNotify {offset; ty; sz; _}, I32 count :: I32 i :: vs' ->
+        let mem = memory frame.inst (0l @@ e.at) in
+        let addr = I64_convert.extend_i32_u i in
+        (try
+          check_align addr ty sz e.at;
```

Comment:
Shouldn't this symmetrically require `shared` as well?

Reply:
Currently we allow `notify` on unshared memory. It's only `wait` that's disallowed.